lib/commands/utility.go

```diff
@@ -1,19 +1,189 @@
 package commands
 
-import "github.com/urfave/cli/v2"
+import (
+	"fmt"
+
+	"github.com/fatih/color"
+	"github.com/kionsoftware/kion-cli/lib/helper"
+	"github.com/kionsoftware/kion-cli/lib/kion"
+	"github.com/kionsoftware/kion-cli/lib/structs"
+	"github.com/urfave/cli/v2"
+)
 
 // FlushCache clears the Kion CLI cache.
 func (c *Cmd) FlushCache(cCtx *cli.Context) error {
 	return c.cache.FlushCache()
 }
 
 // PushFavorites pushes the local favorites to a target instance of Kion.
-func (c *Cmd) PushFavorites(cCtx *cli.Context) error {
-	// identify target instance of Kion
-	// check version of target instance of Kion
-	// pull existing upstream favorites
-	// deconflict with local version of favorites, id to be pushed
-	// prompt user to confirm what will be pushed where
-	// perform push or bail
-	return nil
+func (c *Cmd) PushFavorites(configPath string) func(cCtx *cli.Context) error {
+	return func(cCtx *cli.Context) error {
+
+		if cCtx.App.Metadata["useFavoritesAPI"].(bool) {
+
+			// get the combined list of favorites from the CLI config and the Kion API
+			apiFavorites, _, err := kion.GetAPIFavorites(c.config.Kion.Url, c.config.Kion.ApiKey)
+			if err != nil {
+				fmt.Printf("Error retrieving favorites from API: %v\n", err)
+				return nil
```

Comment:
This should return the error instead of nil. The command will bubble it all the way up to exit.

Not going to add to all other returns in this file but it needs to happen for them too, returning the error instead of nil.

Reply:
This should still be a non-zero exit. Thoughts?